File: GravityCode/Download/Download truyen/gethtmlfromurl-git-e78ccddd8982df4992a26d85295b91df9386a41c/src/models/LoginInfo.java
package models;

import utils.UrlHandler;

/**
 * login info when get content form website
 * 
 * @author nxcuo
 * @since
 */
public class LoginInfo {
	String pageCode;
	String cookies;

	/**
	 * init to serilization
	 */
	public LoginInfo() {
	}

	/**
	 * init with pageCode
	 * 
	 * @param pageCode
	 */
	public LoginInfo(String pageCode) {
		this.pageCode = pageCode;
	}

	public String getPageCode() {
		return pageCode;
	}

	public void setPageCode(String pageCode) {
		this.pageCode = pageCode;
	}

	public String getCookies() {
		return cookies;
	}

	public void setCookies(String cookies) {
		this.cookies = cookies;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj) {
			return true;
		}
		if (obj == null) {
			return false;
		}
		if (getClass() != obj.getClass()) {
			return false;
		}
		LoginInfo other = (LoginInfo) obj;
		if (pageCode == null) {
			if (other.pageCode != null) {
				return false;
			}
		} else {
			// compare * host pattern interchanges
			if (pageCode.contains("*") || other.pageCode.contains("*")) {
				if (pageCode.contains("*") && !other.pageCode.contains(pageCode.replace("*", ""))) {
					return false;
				}
				if (other.pageCode.contains("*") && !pageCode.contains(other.pageCode.replace("*", ""))) {
					return false;
				}
			} else if (!UrlHandler.normalizeHost(pageCode).equals(UrlHandler.normalizeHost(other.pageCode))) {
				return false;
			}
		}
		return true;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((pageCode == null) ? 0 : pageCode.hashCode());
		return result;
	}
}

File: GravityCode/Download/Download truyen/GetHtmlFromUrl_Python/models/login_info.py
from dataclasses import dataclass

@dataclass
class LoginInfo:
    page_code: str = ""
    cookies: str = ""

    def __eq__(self, other):
        if not isinstance(other, LoginInfo):
            return False
        
        # Mô phỏng logic so sánh host pattern của Java
        p1 = self.page_code
        p2 = other.page_code
        
        if not p1 or not p2:
            return p1 == p2
            
        if "*" in p1 or "*" in p2:
            if "*" in p1 and p1.replace("*", "") not in p2:
                return False
            if "*" in p2 and p2.replace("*", "") not in p1:
                return False
            return True
            
        from utils.url_handler import normalize_host
        return normalize_host(p1) == normalize_host(p2)

    def __hash__(self):
        return hash(self.page_code)
